src/main/java/com/codejoust/main/service/ProblemService.java

```diff
@@ -34,19 +35,25 @@
 @Service
 public class ProblemService {
 
+    private final FirebaseService service;
     private final ProblemRepository repository;
+    private final AccountRepository accountRepository;
     private final List<DefaultCodeGeneratorService> defaultCodeGeneratorServiceList;
     private final Random random = new Random();
     private final Gson gson = new Gson();
-    private static final String PROBLEM_ACCESS_PASSWORD_KEY = "PROBLEM_ACCESS_PASSWORD";
 
     @Autowired
-    public ProblemService(ProblemRepository repository, List<DefaultCodeGeneratorService> defaultCodeGeneratorServiceList) {
+    public ProblemService(FirebaseService service, ProblemRepository repository, AccountRepository accountRepository,
+                          List<DefaultCodeGeneratorService> defaultCodeGeneratorServiceList) {
+        this.service = service;
         this.repository = repository;
+        this.accountRepository = accountRepository;
         this.defaultCodeGeneratorServiceList = defaultCodeGeneratorServiceList;
     }
 
-    public ProblemDto createProblem(CreateProblemRequest request) {
+    public ProblemDto createProblem(CreateProblemRequest request, String token) {
+        String uid = service.verifyToken(token);
```

Comment:
Here since it's creating a problem, it doesn't have an existing owner UID yet, so when the problem is created it will assign the owner as whatever UID is decoded from the given token

Reply:
So the UID is only retrieved once a problem is created? Or once a request to the backend is made?

(Just want to make sure I understand this setup.)

Reply:
I think the frontend always has access to the UID when logged in, but in order to be secure, we have them send their token instead - if they are who they claim to be, the token will be decrypted into their UID anyways, and this way it prevents someone from pretending to be another person by using someone else's UID

When creating a problem, it'll save the descrypted UID with that problem so then in the future when editing that problem, it can compare tokens against that UID to make sure the editor is the same user as the original